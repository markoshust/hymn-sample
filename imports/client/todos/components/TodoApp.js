import React from 'react';
import AddTodo from './AddTodo';
import TodoList from '../containers/TodoList';
import Footer from './Footer';

const TodoApp = () => (
  <div>
    <AddTodo />
    <TodoList />
    <Footer />
  </div>
);

export default TodoApp;
