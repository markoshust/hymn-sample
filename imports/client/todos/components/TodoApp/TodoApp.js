import React from 'react';
import AddTodo from '../../containers/AddTodo';
import TodoList from '../../containers/TodoList';
import FilterLinks from '../FilterLinks';

const TodoApp = () => (
  <div>
    <AddTodo />
    <TodoList />
    <FilterLinks />
  </div>
);

export default TodoApp;
