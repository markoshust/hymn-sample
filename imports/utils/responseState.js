const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
