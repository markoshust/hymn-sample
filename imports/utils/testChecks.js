import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
