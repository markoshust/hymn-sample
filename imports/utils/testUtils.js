import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
