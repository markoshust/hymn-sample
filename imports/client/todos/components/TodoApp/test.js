jest.mock('../../containers/AddTodo');
jest.mock('../../containers/TodoList');
jest.mock('../../containers/FilterLink');

/* eslint-disable import/imports-first */
import React from 'react';
import { renderCheck } from 'utils/testChecks';
import TodoApp from './TodoApp';
/* eslint-enable import/imports-first */

describe('<TodoApp />', () => {
  it('renders correctly', () => {
    renderCheck(
      <TodoApp />
    );
  });
});
