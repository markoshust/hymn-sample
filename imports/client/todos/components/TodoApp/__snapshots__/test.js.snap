exports[`<TodoApp /> renders correctly 1`] = `
<div>
  <p>
    Show:
     
    , 
    , 
  </p>
</div>
`;
